zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/FlownodeInstanceQueryController.java

```diff
@@ -51,11 +52,18 @@ private ResponseEntity<FlownodeInstanceSearchQueryResponse> search(
               .search(query);
       return ResponseEntity.ok(
           SearchQueryResponseMapper.toFlownodeInstanceSearchQueryResponse(result));
-    } catch (final Throwable e) {
+    } catch (final ValidationException e) {
```

Comment:
I am wondering what can throw `jakarta.validation.ValidationException` in the try block 🤔 , are we using some constraint annotations ?

Reply:
Hey @houssain-barouni, I gonna answer this one as I am proposed this in the past 

Possible Sources of ValidationException in the Code:

	1.	Bean Validation on Request Objects - (@Valid, @NotNull, etc.) can trigger validation before the method is invoked. However, in this code, there is no explicit validation annotation like @Valid on the query parameter of the searchFlownodeInstances method. This reduces the likelihood that the exception is triggered directly by this controller method.
	2.	Manual Validation within Services: - The most likely place where the ValidationException could be thrown is within the InstanceServices class or any related service that handles the Query object.
	3.	Search Query Validation: If Query (or its related data structures) has any validation annotations and those are programmatically validated within the service layer, a ValidationException could be thrown if the data does not meet the expected constraints.

 The `ValidationException` is caught to handle scenarios where the query or any of its fields fail these validation checks, ensuring the client receives a meaningful error response.
And the goal is not return only 500 Status code, in case of any error, but separate the validation code between Request/Internal server.

Reply:
The caller will send a query that will be validated either by the annotations generated in the API data model or through manual validations, both of which occur outside this try block. If any other validation fails within this try block, it is beyond the caller’s control, meaning they cannot adjust the query to make it work. So, the 400 error in this case is kind of useless.